### PR TITLE
Unstarded story state

### PR DESCRIPTION
**Pull request by rubynho**
# Set an estimated story to unstarted automatically

### Feature
Set an estimable story as unstarted if it gets estimated and set it back to unscheduled if it gets unestimated.

An estimable story can't have a state of unstarted while it's not estimated and can't have a state of unscheduled while it's estimated.

#### Estimable story types?
So far only the `feature` stories are estimable.

#### How to test?
In order to test you should check if when you estimate an estimable story (feature one) it will change the state to `unstarted` and when unestimate it will set the state back to `unscheduled`. Make sure the stories the are not estimable (chore, bug and release ones) don't have the same behavior.

### CODE REVIEW

File: app/models/story.rb [hunk 1]
```diff
@@ -104,4 +106,12 @@ def saved_changes
   def saved_changes?
     documents_attributes_was != documents_attributes || super
   end
+
+  def self.can_be_estimated?(story_type)
+    ESTIMABLE_STORY_TYPES.include? story_type.to_s
+  end
+
+  def can_be_estimated?
+    ESTIMABLE_STORY_TYPES.include? story_type
```
Comment (jonatasrancan) [inline app/models/story.rb]: Faz assim:

```ruby
self.class.can_be_estimated(story_type)
```

Comment (rubynho) [reply app/models/story.rb]: Boa!

File: spec/operations/story_operations_spec.rb [hunk 2]
```diff
@@ -127,9 +145,29 @@
...
+      end
+
+      context "when underestimate #{story_type} story" do
+        let(:params) { { 'estimate' => nil, 'story_type' => story_type } }
+
+        it 'sets the story state as unscheduled' do
+          story.attributes = { estimate: 1, state: 'unstarted' }
+          subject.call
+          expect(story.state).to eq('unscheduled')
+        end
```
Comment (talyssonoc) [inline spec/operations/story_operations_spec.rb]: Can we add a test for the following scenario:

- we have a _started_ estimable story with an estimate of 1
- we change only the estimate of this story to 2
- we verify it didn't affect the state, it remains _starter_

Comment (rubynho) [reply spec/operations/story_operations_spec.rb]: Sure!

**Pull request merged**